scripts/dev/newReleaseBranch.sh

```diff
@@ -0,0 +1,681 @@
+/.*version.minor.*$/s//        <version.minor>${versionMinor}<\/version.minor>/
+/.*version.revision.*$/s//        <version.revision>${versionRevision}<\/version.revision>/
+wq
+EOF
+  fi
+}
+
+setNextReleaseVersion() {
+  local version="${1}"
+
+  echo "${prefix}    set - RELEASE_VERSION=${version}"
+  ed -s .travis.yml &>${loggingOut} << EOF
+/.*- RELEASE_VERSION.*$/s//    - RELEASE_VERSION=${version}/
+wq
+EOF
+}
+
+setStartWithRealVersion() {
+  local version="${1}"
+
+  echo "${prefix}    set # ... start with real version ${version}"
+  ed -s .travis.yml &>${loggingOut} << EOF
+/.*start with real version.*$/s//    # Release version has to start with real version (${version}-....) for the docker image to build successfully./
+wq
+EOF
+}
+
+setNextDevelopmentVersion() {
+  local version="${1}"
+
+  echo "${prefix}    set - DEVELOPMENT_VERSION=${version}"
+  ed -s .travis.yml &>${loggingOut} << EOF
+/.*- DEVELOPMENT_VERSION.*$/s//    - DEVELOPMENT_VERSION=${version}/
+wq
+EOF
+}
+
+setVersionInPackaging() {
+  local version="${1}"
+  local branchType="${2}"
+  local projectType="${3}"
+
+  if [[ "${projectType}" == "Packaging" ]]
+  then
+    if [[ "${branchType}" == "HotFix" ]]
+    then
+      setNextReleaseVersion     "${version}.1"
+      setNextDevelopmentVersion "${version}.2-SNAPSHOT"
+    else
+      setStartWithRealVersion   "${version}"
+      setNextReleaseVersion     "${version}-A1"
+      setNextDevelopmentVersion "${version}-SNAPSHOT"
+    fi
+  fi
+}
+
+setUpstreamVersionsInCommunityPackaging() {
+  local nextCommunityRepoVersion="${1}"
+  local nextEnterpriseShareVersion="${2}"
+
+  if [[ `getCurrentProject` == "acs-community-packaging" ]]
+  then
+    echo "${prefix}    set parent pom <version> ${nextCommunityRepoVersion}"
+    echo "${prefix}    set <dependency.alfresco-community-repo.version> ${nextCommunityRepoVersion}"
+    echo "${prefix}    set <dependency.alfresco-community-share.version> ${nextEnterpriseShareVersion}"
+    # Do not change the <dependency.acs-packaging.version> as we will need a version of the share distribution that
+    # exists, if we want to do a test build. acs-packaging will update it when a release is made.
+    ed -s pom.xml &>${loggingOut} << EOF
+/        <version>.*$/s//        <version>${nextCommunityRepoVersion}<\/version>/
+/.*dependency.alfresco-community-repo.version.*$/s//        <dependency.alfresco-community-repo.version>${nextCommunityRepoVersion}<\/dependency.alfresco-community-repo.version>/
+/.*dependency.alfresco-community-share.version.*$/s//        <dependency.alfresco-community-share.version>${nextEnterpriseShareVersion}<\/dependency.alfresco-community-share.version>/
+wq
+EOF
+  fi
+}
+
+incrementSchema() {
+  local schemaMultiple="${1}"
+
+  if [[ `getCurrentProject` == "alfresco-community-repo" && "${schemaMultiple}" -gt 0 ]]
+  then
+    local schema=`getSchema`
+    schema=`echo "(${schema} / ${schemaMultiple} + 1) * ${schemaMultiple}" | bc`
```

Comment:
What's happening here? If schemaMultiple is for example 100 (ServicePack) and current 
```
# Schema number
version.schema=17000
```
then it goes down to (17000/101) * 100 = ~16831
Are the multipliers in right order? I don't have much of knowledge about the release process itself or versioning, so it might be something obvious. If it is, then possibly a comment would be nice, as I have no idea what the schema multiplication is for (aside from changing version of schema to avoid clashes, but why such an operation is needed I don't have the slightest clue).